src/utils/DefaultAccessControlLateInit.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import "../interfaces/utils/IDefaultAccessControl.sol";
+
+/// @notice This is a default access control with 3 roles:
+///
+/// - ADMIN: allowed to do anything
+/// - ADMIN_DELEGATE: allowed to do anything except assigning ADMIN and ADMIN_DELEGATE roles
+/// - OPERATOR: low-privileged role, generally keeper or some other bot
+contract DefaultAccessControlLateInit is IDefaultAccessControl, AccessControlEnumerable {
+    error AddressZero();
+    error Forbidden();
+    error Initialized();
+
+    bool public initialized;
```

Comment:
I think it's better to move `initialized` variable and its logic in the `Vault` contract, near to `isPaused`, `isPublic`